SwiftCompilerSources/Sources/Basic/SourceLoc.swift

```diff
@@ -0,0 +1,73 @@
+//===--- SourceLoc.swift - SourceLoc bridiging utilities ------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+public struct SourceLoc {
```

Comment:
I always found it confusing that we have multiple data types to describe a source location (in C++): `SourceLoc` and `SILLocation`.
I think we can do better on the swift side. We should unify `AST.SourceLoc` and `SIL.Location` and have only a single data type for that.

Reply:
`SourceLoc` and `SILLocation` are very different type. `SourceLoc` is basically just a pointer, whereas `SILLocation` is a much more complex object. I'm not sure how we could unify them...